.gitignore

```diff
@@ -0,0 +1,42 @@
+# Compiled object files
+*.o
+
+# Executables
+*.exe
+*.out
+
+# Compiled libraries
+*.a
+*.so
+
+# Disable for now
+*.cmake
+
+# Build directory
+/build/
+/cmake-build-debug/
+
+# External directoy
+/external/
+
+# Doxygen output path
+/docs/
+
+# Dependency files
+*.d
+
+# Editor/IDE specific files
+.vscode/
+.idea/
+
+# Dont track sonnar files
+.scannerwork/
+
+# Documentation files (Doxygen)
+/html/
+/latex/
+/rtf/
+/man/
+/xml/
+/docbook/
+/sqlite3/
```

Comment:
Those files would be generated by doxyfile if certain Generate flags were on YES (like GENERATE_HTML), currently those are off, thought of adding them to the gitignore just in case since I'm not really sure if there should be track of them. I'll erease them since the flags are off anyway.

Reply:
Yes, let's simplify the `.gitignore` 😃